rust/src/chain_crypto/algorithms/ed25519_derive.rs

```diff
@@ -0,0 +1,121 @@
+use crate::chain_crypto::key::{
+    AsymmetricKey, AsymmetricPublicKey, PublicKeyError, SecretKeyError, SecretKeySizeStatic,
+};
+use crate::chain_crypto::sign::{SignatureError, SigningAlgorithm, Verification, VerificationAlgorithm};
+
+use ed25519_bip32 as i;
+use ed25519_bip32::{XPrv, XPub, XPRV_SIZE, XPUB_SIZE};
+use rand_os::rand_core::{CryptoRng, RngCore};
+
+/// Ed25519 BIP32 Signature algorithm
+pub struct Ed25519Bip32;
+
+impl From<i::PrivateKeyError> for SecretKeyError {
+    fn from(v: i::PrivateKeyError) -> Self {
+        match v {
+            i::PrivateKeyError::HighestBitsInvalid => SecretKeyError::StructureInvalid,
+            i::PrivateKeyError::LowestBitsInvalid => SecretKeyError::StructureInvalid,
+            i::PrivateKeyError::LengthInvalid(_) => SecretKeyError::SizeInvalid,
+        }
+    }
+}
+
+impl From<i::PublicKeyError> for PublicKeyError {
+    fn from(v: i::PublicKeyError) -> Self {
+        match v {
+            i::PublicKeyError::LengthInvalid(_) => PublicKeyError::SizeInvalid,
+        }
+    }
+}
+
+impl AsymmetricPublicKey for Ed25519Bip32 {
+    type Public = XPub;
+    const PUBLIC_BECH32_HRP: &'static str = "xpub";
+    const PUBLIC_KEY_SIZE: usize = XPUB_SIZE;
+    fn public_from_binary(data: &[u8]) -> Result<Self::Public, PublicKeyError> {
+        let xpub = XPub::from_slice(data)?;
+        Ok(xpub)
+    }
+}
+
+impl AsymmetricKey for Ed25519Bip32 {
+    type Secret = XPrv;
+    type PubAlg = Ed25519Bip32;
+
+    const SECRET_BECH32_HRP: &'static str = "xprv";
+
+    fn generate<T: RngCore + CryptoRng>(mut rng: T) -> Self::Secret {
+        let mut priv_bytes = [0u8; XPRV_SIZE];
+        rng.fill_bytes(&mut priv_bytes);
+        XPrv::normalize_bytes_force3rd(priv_bytes)
```

Comment:
Why did this get changed from `normalize_bytes` to `normalize_bytes_force3rd`?

It seems this change originates from this upstream commit (https://github.com/input-output-hk/chain-libs/commit/0295b6db5fea512838a957739cc33746c6d0da88), and the code is only partially ported into this PR. I'm not sure if it's safe to replace just this part and not make the rest of the code changes from this commit.